tests/disk/test_disk_exhaustion.py

```diff
@@ -0,0 +1,210 @@
+
+    return route_targets, ifaces
+
+
+def parse_rif_counters(output_lines):
+    """
+    Parse the output of "show interfaces counters rif" command
+    """
+
+    header_line = ''
+    separation_line = ''
+    separation_line_number = 0
+    for idx, line in enumerate(output_lines):
+        if line.find('----') >= 0:
+            header_line = output_lines[idx - 1]
+            separation_line = output_lines[idx]
+            separation_line_number = idx
+            break
+
+    try:
+        positions = parse_column_positions(separation_line)
+    except Exception:
+        logger.error('Possibly bad command output')
+        return {}
+
+    headers = []
+    for pos in positions:
+        header = header_line[pos[0]:pos[1]].strip().lower()
+        headers.append(header)
+
+    if not headers:
+        return {}
+
+    results = {}
+    for line in output_lines[separation_line_number + 1:]:
+        portstats = []
+        for pos in positions:
+            portstat = line[pos[0]:pos[1]].strip()
+            portstats.append(portstat)
+
+        intf = portstats[0]
+        results[intf] = {}
+        for idx in range(1, len(portstats)):  # Skip the first column interface name
+            results[intf][headers[idx]] = portstats[idx]
+
+    return results
+
+
+def construct_packet_and_get_params(duthost, ptfadapter, tbinfo):
+    """
+    Construct data packet and get related params
+    """
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+
+    # generate peer_ip and port channel pair, be like:[("10.0.0.57", "PortChannel0001")]
+    peer_ip_pc_pair = [(pc["peer_addr"], pc["attachto"]) for pc in mg_facts["minigraph_portchannel_interfaces"]
+                       if
+                       ipaddress.ip_address(pc['peer_addr']).version == 4]
+
+    pc_ports_map = {pair[1]: mg_facts["minigraph_portchannels"][pair[1]]["members"] for pair in
+                    peer_ip_pc_pair}
+
+    if len(mg_facts["minigraph_interfaces"]) >= 2:
+        # generate peer_ip and interfaces pair,
+        # be like:[("10.0.0.57", ["Ethernet48"])]
+        peer_ip_ifaces_pair = [(intf["peer_addr"], [intf["attachto"]]) for intf in mg_facts["minigraph_interfaces"]
+                               if
+                               ipaddress.ip_address(intf['peer_addr']).version == 4]
+
+    else:
+        # generate peer_ip and interfaces(port channel members) pair,
+        # be like:[("10.0.0.57", ["Ethernet48", "Ethernet52"])]
+        peer_ip_ifaces_pair = [(pair[0], mg_facts["minigraph_portchannels"][pair[1]]["members"]) for pair in
+                               peer_ip_pc_pair]
+
+    # use first port of first peer_ip_ifaces pair as input port
+    # all ports in second peer_ip_ifaces pair will be output/forward port
+    ptf_port_idx = mg_facts["minigraph_ptf_indices"][peer_ip_ifaces_pair[0][1][0]]
+    # Some platforms do not support rif counter
+    try:
+        rif_counter_out = parse_rif_counters(
```

Comment:
The lib already has a help function to parse the tabulated command output. You can refer to: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/devices/sonic.py#L1408

Reply:
I will check and improve, thanks!